### PR TITLE
Closes #133 update lintr configurations with central {admiraldev} file

### LINKED ISSUE #133: Update lintr configurations to use central `{admiraldev}` file
The `{admiral}` team would like to enhance the way we perform lint checks in our packages.

Pull request https://github.com/pharmaverse/admiraldev/pull/507 will, once merged, introduce a way to set up custom `{lintr}` configurations in `{admiraldev}` which can then used by all other packages in the `{admiral}` ecosystem. If the configurations are updated in `{admiraldev}`, they will trickle down to the extension packages, unless they are overridden (see below). Currently the configurations we have chosen are:

- line length: 100 (same as previously)
- object usage: NULL (same as previously)
- cyclomatic complexity: 20 (same as previously)
- indentation: NULL (same as previously)
- undesirable functions: `stop()`, `warning()`, `message()`, `ifelse()`, `grepl()` 🆕

To implement the default configurations, simply make a PR for this issue where you:

- Install the development version of `{admiraldev}`
- Rename the `.lint` config file to `.lintr.R`
- Replace the contents of the file with:
```
library(lintr)

source(system.file("lintr/linters.R", package = "admiraldev"))

linters <- admiral_linters()

exclusions <- list(
  "R/data.R" = Inf,
  "inst" = list(undesirable_function_linter = Inf),
  "vignettes" = list(undesirable_function_linter = Inf)
)
```
- Run `lintr::lint_package()` to check the linters run as expected
- Fix any new issues with lints; it's likely you'll have been using at least a few of the now undesirable functions.
- Update `.Rbuildignore` to ignore new `.lintr.R `file.

If you have a legitimate reason to override any of the default configurations, you can just pass arguments to `admiral_linters()` - see the [function definition](https://github.com/pharmaverse/admiraldev/blob/lintr_testing/inst/lintr/linters.R) (link will work once PR is merged) for details on how to do so.

### PR DESCRIPTION
**Pull request by Fanny-Gautier**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build pharmaverseadam site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new datasets occur on the "[Reference](https://pharmaverse.github.io/pharmaverseadam/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/create_adams_data.R [hunk 1]
```diff
@@ -173,16 +173,20 @@ run_template <- function(tp) {
 
       # add PARAM and PARAMCD details
       # check if data contains PARAMCD/PARAM variables
-      param_col <- names(data)[grepl("PARAM$", names(data))]
-      paramcd_col <- names(data)[grepl("PARAMCD", names(data))]
+      param_col <- names(data)[stringr::str_detect("PARAM$", names(data))]
```
[the remark this answers is not in the document]
Comment (Fanny-Gautier) [reply data-raw/create_adams_data.R]: Good catch! Thank you.

**Comment by manciniedoardo** [comment]
@Fanny-Gautier 

I think the lint check is not working remotely because the .lintr.R file now assumes admiraldev is an installed package. please try adding admiraldev to the Suggests: field in the DESCRIPTION file and also in the upstream_repos field of staged_dependencies.yaml  so that it is installed prior to the CICD checks running.  See [here](https://github.com/pharmaverse/admiraldev/blob/main/staged_dependencies.yaml) for an example. This also explains why you saw everything working fine locally, when admiraldev was already installed.
